src/connection.ts

```diff
@@ -376,6 +387,7 @@ export class Connection {
 
 export type ListenersParams = {
   metadata_update: MetadataUpdateListener
+  publish_confirm: PublishConfirmListener
```

Comment:
These two listeners are not mandatory 
```suggestion
  metadata_update?: MetadataUpdateListener
  publish_confirm?: PublishConfirmListener
```